storage/badger/dkg_state.go

```diff
@@ -219,21 +224,74 @@ func (ds *RecoverablePrivateBeaconKeyStateMachine) RetrieveMyBeaconPrivateKey(ep
 	return
 }
 
+// CommitMyBeaconPrivateKey commits the previously inserted random beacon private key for an epoch.
+// Effectively, this method transitions the state machine into the [flow.RandomBeaconKeyCommitted] state if the current state is [flow.DKGStateCompleted].
+// Caller needs to supply the [flow.EpochCommit] which is an evidence that the key has been indeed included for the given epoch.
+// No errors are expected during normal operations.
```

Comment:
```suggestion
// If the current state is already [flow.RandomBeaconKeyCommitted], this function is a no-op regardless of input.
// No errors are expected during normal operations.
```
Describing the behaviour on line 238.

Reply:
regarding Jordan's suggestion:
> If the current state is already [flow.RandomBeaconKeyCommitted], this function is a no-op regardless of input.

This is probably the most minimal implementation, but I am not sure it would be the most robust. I think there are two non-happy path scenarios I think we should consider for the `CommitMyBeaconPrivateKey`:
1. A random beacon key was previously committed (i.e. we are in the `flow.RandomBeaconKeyCommitted` state). If there is a call to `CommitMyBeaconPrivateKey` with an `flow.EpochCommit` that is _inconsistent_ with the committed key, we have a clear failure condition. 
2. A random beacon key was previously committed (i.e. we are in the `flow.RandomBeaconKeyCommitted` state). The state machine receives a call to `CommitMyBeaconPrivateKey` with an `flow.EpochCommit` that is _consistent_ with the committed key. Here, we could either return an exception and say that repeated calls are in principle not allowed. Alternatively, we can treat this call as confirming information that the state machine already has and ignore this call (as information is idempotent). The latter is my preference. 

What makes me nervous is taking the response strategy for 2. and also applying it to 1., because we would be proceeding in a clear failure case. I think it would be comparatively easy to be more strict in this case.